apps/typegpu-docs/src/content/examples/simple/function-visualizer/index.ts

```diff
@@ -0,0 +1,532 @@
+    properties: propertiesBuffer,
+  });
+
+  renderBackgroundPassDescriptor.colorAttachments[0].view = context
+    .getCurrentTexture()
+    .createView();
+
+  const encoder = device.createCommandEncoder({ label: 'Render encoder' });
+
+  const pass = encoder.beginRenderPass(renderBackgroundPassDescriptor);
+  pass.setPipeline(renderBackgroundPipeline);
+  pass.setBindGroup(0, root.unwrap(renderBindGroup));
+  pass.draw(4, 2);
+  pass.end();
+
+  const commandBuffer = encoder.finish();
+  device.queue.submit([commandBuffer]);
+}
+
+function runRenderPass() {
+  renderPassDescriptor.colorAttachments[0].view = context
+    .getCurrentTexture()
+    .createView();
+
+  const encoder = device.createCommandEncoder({ label: 'Render encoder' });
+
+  const pass = encoder.beginRenderPass(renderPassDescriptor);
+
+  initialFunctions.forEach((_, i) => {
+    const renderBindGroup = root.createBindGroup(renderLayout, {
+      lineVertices: lineVerticesBuffers[i],
+      properties: propertiesBuffer,
+      color: drawColorBuffers[i],
+    });
+    pass.setPipeline(renderPipeline);
+    pass.setBindGroup(0, root.unwrap(renderBindGroup));
+    // call our vertex shader 2 times per point drawn
+    pass.draw(properties.interpolationPoints * 2);
+  });
+  pass.end();
+
+  device.queue.submit([encoder.finish()]);
+}
+
+// Helper definitions
+
+function fromHex(hex: string) {
+  const r = Number.parseInt(hex.slice(1, 3), 16);
+  const g = Number.parseInt(hex.slice(3, 5), 16);
+  const b = Number.parseInt(hex.slice(5, 7), 16);
+
+  return d.vec4f(r / 255.0, g / 255.0, b / 255.0, 1.0);
+}
+
+async function tryRecreateComputePipeline(
+  functionCode: string,
+): Promise<GPUComputePipeline> {
+  const codeToCompile = functionCode === '' ? '0' : functionCode;
+
+  const computeShaderCode = createComputeShaderCode(codeToCompile);
+
+  device.pushErrorScope('validation');
+  const computeShaderModule = device.createShaderModule({
+    label: `Compute function points shader module for f(x) = ${codeToCompile}`,
+    code: computeShaderCode,
+  });
+  const error = await device.popErrorScope();
+  if (error) {
+    throw new Error(`Function f(x) = '${codeToCompile}' is invalid.`);
+  }
+
+  const computePipeline = device.createComputePipeline({
+    label: 'Compute function points pipeline',
+    layout: device.createPipelineLayout({
+      bindGroupLayouts: [root.unwrap(computeLayout)],
+    }),
+    compute: {
+      module: computeShaderModule,
+    },
+  });
+
+  return computePipeline;
+}
+
+function queuePropertiesBufferUpdate() {
+  properties.inverseTransformation = mat4.inverse(properties.transformation);
+  propertiesBuffer.write(properties);
+}
+
+// Canvas controls
+
+let lastPos: number[] | null = null;
+
+canvas.onmousedown = (event) => {
+  lastPos = [event.offsetX, event.offsetY];
+};
+
+canvas.onmouseup = (_) => {
+  lastPos = null;
+};
+
+canvas.onmousemove = (event) => {
+  if (lastPos == null) {
```

Comment:
Let's use strict equals when we can 🫡 
```suggestion
  if (lastPos === null) {
```